### ISSUE TITLE
Add support to enable `migrateEncryption` option

### ISSUE THREAD

**Issue by jlouisfoster** [hashicorp/packer-plugin-vsphere#467]
#### Summary

When attempting to add `"migrate.encryptionMode" = "required"` via `configuration_parameters` option in `.pkr.hcl` - this setting does not get added to the `.vmx` file; thus the change does not occur on the template/vm.

#### Reproduction Steps

Update `configuration_parameters` with `"migrate.encryptionMode" = "required"` and validate if Encrypted vMotion is set to required.

### Plugin and Packer version

Packer v1.10.2

### Simplified Packer Buildfile

```hcl
  network_adapters {
    network      = var.vm_network
    network_card = var.network_card
  }

  configuration_parameters = {
    "tools.guest.desktop.autolock" = "FALSE"
    "log.keepOld" = "15"
    "migrate.encryptionMode" = "required"
  }
```

### Operating system and Environment details

Windows Server 2022

### Log Fragments and crash.log files

No error in logs

**Comment by tenthirtyam** [comment]
Try `"migrateEncryption" = "required"`.

**Comment by jlouisfoster** [comment]
Hi,

Thank you for the quick reply :) 

I tried with that option but still no luck...

**Comment by tenthirtyam** [comment]
No worries. I took a look at the code last night and have an idea why this is not working and a possible fix I plan on testing in my available time next week.

**Comment by jlouisfoster** [comment]
Thank you!

I really appreciate the support

**Comment by tenthirtyam** [comment]
Seems that it's one that is ignored by [`extraConfig`](https://dp-downloads.broadcom.com/api-content/apis/API_VWSA_001/8.0U3/html/ReferenceGuides/vim.vm.ConfigSpec.html#extraConfig).

Similar issue [in the sdk](https://github.com/vmware/govmomi/issues/3493).

Ryan

**Comment by jlouisfoster** [comment]
Hmm so I’m guessing - this is a fix that is needed at the VMware level?

**Comment by tenthirtyam** [comment]
No, actually if would need to be an enhancement request for this plugin to allow [`migrateEncryption`](https://dp-downloads.broadcom.com/api-content/apis/API_VWSA_001/8.0U3/html/ReferenceGuides/vim.vm.ConfigSpec.html#extraConfig:~:text=API%20Release%208.0.3.0-,migrateEncryption*,-xsd%3Astring) to be set.

Since [`migrateEncryption`](https://dp-downloads.broadcom.com/api-content/apis/API_VWSA_001/8.0U3/html/ReferenceGuides/vim.vm.ConfigSpec.html#extraConfig:~:text=API%20Release%208.0.3.0-,migrateEncryption*,-xsd%3Astring) is explicitly configurable through in the `ConfigSpec` object it is silently ignored when sent in the [extraConfig](https://dp-downloads.broadcom.com/api-content/apis/API_VWSA_001/8.0U3/html/ReferenceGuides/vim.vm.ConfigSpec.html#extraConfig).

**Ryan Johnson**
Distinguished Engineer, VMware by Broadcom

**Comment by tenthirtyam** [comment]
Converted to an `enhancement`.

**Comment by jlouisfoster** [comment]
Is it possible we can modify the extraconfig via packer?

**Comment by tenthirtyam** [comment]
> Is it possible we can modify the extraconfig via packer?

That's what `configuration_parameters` allows for but if there is a conflict, it will be ignored.

I submitted the following pull request to add some additional logging and update the docs in that regard.

Take a look at the log output.

https://github.com/hashicorp/packer-plugin-vsphere/pull/468

**Comment by jlouisfoster** [comment]
Really appreciate it tenthiryam - sorry for all the questions; it's one of my first enhancement requests.
Will that PR (https://github.com/hashicorp/packer-plugin-vsphere/pull/468) fix that issue once approved or just provide more logs on how it could be fixed?

**Comment by tenthirtyam** [comment]
> Will that PR fix that issue once approved or just provide more logs on how it could be fixed?

It will just log that the item was ignored and include an update in the docs.

**Comment by jlouisfoster** [comment]
So the only method is to manually make this change - automating it with packer is not a possibility?

**Comment by tenthirtyam** [comment]
> So the only method is to manually make this change - automating it with packer is not a possibility?

Correct.

**Comment by jlouisfoster** [comment]
Will look into that, thank you!

**Comment by jlouisfoster** [comment]
Would you happen to have an example in the docs of setting up a provisioner with powercli for packer?

**Comment by tenthirtyam** [comment]
Marking as closed, not planned.
